src/core/linalg/src/sparse/4C_linalg_map.hpp

```diff
@@ -45,122 +64,289 @@ namespace Core::LinAlg
-    int MaxAllGID() const { return map_->MaxAllGID(); }
+    //! Returns the maximum global ID across the entire map.
+    int MaxAllGID() const
+    {
+      return visit_variant([](const auto& map) { return map.MaxAllGID(); });
+    }
 
     //! Returns the minimum global ID across the entire map.
-    int MinAllGID() const { return map_->MinAllGID(); }
+    int MinAllGID() const
+    {
+      return visit_variant([](const auto& map) { return map.MinAllGID(); });
+    }
 
     //! Returns local ID of global ID, return -1 if not found on this processor.
-    int LID(int GID) const { return map_->LID(GID); }
+    int LID(int GID) const
+    {
+      return visit_variant([&](const auto& map) { return map.LID(GID); });
+    }
 
     //! Returns true if this and Map have identical point-wise structure
-    bool PointSameAs(const Epetra_Map& Map) const { return map_->PointSameAs(Map); }
-    bool PointSameAs(const Epetra_BlockMap& Map) const { return map_->PointSameAs(Map); }
+    bool PointSameAs(const Epetra_Map& Map) const
+    {
+      return visit_variant([&](const auto& map) { return map.PointSameAs(Map); });
+    }
+
+    //! Returns true if this and Map have identical point-wise structure
+    bool PointSameAs(const Epetra_BlockMap& Map) const
+    {
+      return visit_variant([&](const auto& map) { return map.PointSameAs(Map); });
+    }
 
     //! Returns the processor IDs and corresponding local index value for a given list of global
-    //! indices.
+    //! indices
     int RemoteIDList(int NumIDs, int* GIDList, int* PIDList, int* LIDList) const
     {
-      return map_->RemoteIDList(NumIDs, GIDList, PIDList, LIDList);
+      return visit_variant(
+          [&](const auto& map) { return map.RemoteIDList(NumIDs, GIDList, PIDList, LIDList); });
     }
 
+    //! Returns the processor IDs and corresponding local index value for a given list of global
+    //! indices
     int RemoteIDList(
         int NumIDs, const int* GIDList, int* PIDList, int* LIDList, int* SizeList) const
     {
-      return map_->RemoteIDList(NumIDs, GIDList, PIDList, LIDList, SizeList);
+      return visit_variant([&](const auto& map)
+          { return map.RemoteIDList(NumIDs, GIDList, PIDList, LIDList, SizeList); });
     }
-    int MinMyGID(void) const { return map_->MinMyGID(); }
 
-    //! Access function for Epetra_Comm communicator.
-    const Epetra_Comm& EpetraComm() const { return map_->Comm(); }
+    //! Returns the minimum global ID owned by this processor.
+    int MinMyGID(void) const
+    {
+      return visit_variant([](const auto& map) { return map.MinMyGID(); });
+    }
 
-    MPI_Comm Comm() const { return Core::Communication::unpack_epetra_comm(map_->Comm()); }
+    //! Access function for Epetra_Comm communicator.
+    MPI_Comm Comm() const
+    {
+      return Core::Communication::unpack_epetra_comm(
+          visit_variant([](const auto& map) -> const Epetra_Comm& { return map.Comm(); }));
```

Comment:
This call was introduced in #788 and requires the inclusion of the expensive `4C_comm_mpi_utils.hpp` header. Which increased the compile time a lot. Since you are working on this file anyway, could you move this function to the implementation and remove the header-in-header inclusion?